xapian-core/weight/tfidfweight.cc

```diff
@@ -141,14 +141,18 @@ double
 TfIdfWeight::get_idfn(Xapian::doccount termfreq, char c) const
 {
     double N = 1.0;
-    if (c != 'n') N = get_collection_size();
+    if (c != 'n' && c != 'f') N = get_collection_size();
     switch (c) {
 	case 'n':
 	    return 1.0;
 	case 'p':
 	    // All documents are indexed by the term
 	    if (N == termfreq) return 0;
 	    return log((N - termfreq) / termfreq);
+	case 'f':
+	    return (1.0/termfreq);
+    	case 's':
```

Comment:
`xapian-check-patch` says:

```
$ git diff atreus/master | ./xapian-maintainer-tools/xapian-check-patch 
xapian-core/weight/tfidfweight.cc:154: added/changed line has space before tab: +    	case 's':
Files patched:	4
Lines added:	32
Lines removed:	9
```

which means that line need re-indenting to match the others around it (tab first, then it shouldn't need spaces).